app/fruits/page.tsx

```diff
@@ -10,34 +10,29 @@ export const metadata = {
 
 export default function FruitsPage() {
   // get the cookie from the server
-  const fruitsCookie = cookies().get('fruitsCookie');
+  const fruitComments = parseJson(getCookie('fruitComments'));
 
-  // create a default value if cooke doesn't exist
-  let fruitsCookieParsed: CookieValue = [];
+  const currentComments = Array.isArray(fruitComments) ? fruitComments : [];
 
-  if (fruitsCookie) {
-    fruitsCookieParsed = JSON.parse(fruitsCookie.value);
-  }
+  const fruitsWithComments = fruits.map((fruit) => {
+    const fruitWithComment = { ...fruit, comment: '' };
 
-  const fruitsWithStars = fruits.map((fruit) => {
-    const fruitWithStars = { ...fruit, stars: 0 };
-
-    // i read the cookie and find the fruit
-    const fruitInCookie = fruitsCookieParsed.find(
+    // read the cookie and find the fruitNote
+    const fruitInCookie = currentComments.find(
       (fruitObject) => fruit.id === fruitObject.id,
     );
 
-    // if find the fruit i update the amount of stars
+    // if find the fruit update the note
     if (fruitInCookie) {
```

Comment:
this `if` and the `const fruitWithComment = { ...fruit, comment: '' };`  can probably be avoided with the following pattern:

```js
const matchingFruitFromCookie = currentComments.find(
  (fruitObject) => fruit.id === fruitObject.id,
);
return {
  ...fruit,
  comment: matchingFruitFromCookie.comment || '',
}
```

Reply:
but wait - why even do this? why not just let the `comment` be `undefined`?

that seems like more natural data...

Reply:
ae48d6b